File: app/src/main/java/com/bq/zowi/controllers/GameController.java
package com.bq.zowi.controllers;

import java.lang.reflect.Type;
import rx.Single;

/* JADX INFO: loaded from: classes.dex */
public interface GameController {

    public enum GAME_ID {
        ZOWI_SAYS_GAME_ID,
        MOUTHS_GAME_ID,
        TIMELINE_GAME_ID,
        GAMEPAD_GAME_ID
    }

    Single<Boolean> isFirstPlay(GAME_ID game_id, boolean z);

    <T> Single<T> loadProgress(GAME_ID game_id, Class<T> cls);

    <T> Single<T> loadProgress(GAME_ID game_id, Type type);

    Single<Void> resetGamesProgress();

    Single<Void> saveProgress(GAME_ID game_id, Object obj);
}

File: zowi-core/src/main/kotlin/com/bq/zowi/api/GameController.kt
package com.bq.zowi.api

import io.reactivex.Single

interface GameController {
    enum class GAME_ID {
        ZOWI_SAYS_GAME_ID,
        MOUTHS_GAME_ID,
        TIMELINE_GAME_ID,
        GAMEPAD_GAME_ID
    }

    fun isFirstPlay(gameId: GAME_ID, forceFirst: Boolean): Single<Boolean>
    fun <T> loadProgress(gameId: GAME_ID, clazz: Class<T>): Single<T>
    fun <T> loadProgress(gameId: GAME_ID, type: java.lang.reflect.Type): Single<T>
    fun resetGamesProgress(): Single<Void>
    fun saveProgress(gameId: GAME_ID, data: Any): Single<Void>
}
